pkgs/development/libraries/flutter-engine/default.nix

```diff
@@ -0,0 +1,271 @@
+{ lib
+, stdenv
+, stdenvNoCC
+, hostPlatform
+, callPackage
+, fetchFromGitHub
+, fetchurl
+, writeText
+, ninja
+, gnumake
+, patchelf
+, python3
+, clang-tools
+, pkg-config
+, git
+, gclient-wrapped
+}@pkgs:
+with lib;
+let
+  # Get this from "flutter doctor"
+  version = "857bd6b74c5eb56151bfafe91e7fa6a82b6fee25";
+  sha256 = "sha256-+661KBEcyNj1t0h9rp1kM+hv1DdI/pxxnrJtHihnXyc=";
```

Comment:
how exactly?

<details>
<summary>
flutter doctor
</summary>

```
flutter doctor 
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.3, on NixOS 22.11 (Raccoon) 5.15.76, locale en_US.UTF-8)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
[✗] Linux toolchain - develop for Linux desktop
    ✗ clang++ is required for Linux development.
      It is likely available from your distribution (e.g.: apt install clang), or can be downloaded from https://releases.llvm.org/
    ✗ CMake is required for Linux development.
      It is likely available from your distribution (e.g.: apt install cmake), or can be downloaded from https://cmake.org/download/
    ✗ ninja is required for Linux development.
      It is likely available from your distribution (e.g.: apt install ninja-build), or can be downloaded from
      https://github.com/ninja-build/ninja/releases
    ✗ pkg-config is required for Linux development.
      It is likely available from your distribution (e.g.: apt install pkg-config), or can be downloaded from
      https://www.freedesktop.org/wiki/Software/pkg-config/
[!] Android Studio (not installed)
[✓] Connected device (1 available)
[✓] HTTP Host Availability

! Doctor found issues in 4 categories.
```

</details>

does flutter require a specific flutter-engine?
or can we just use the latest flutter-engine?

Reply:
Manually updating it with `nix-shell -p flutter -c "flutter doctor -v"` and finding the line which says the engine version. However, I plan on explicitly setting the flutter tool to use `nixpkgs`'s version via the arguments with the wrapper in a later PR.

Reply:
We also have to update the `deps.nix` file to match the rev's set in the `DEPS` file in the Flutter Engine GitHub repo.